Bootloader/Src/firmware_update.c

```diff
@@ -358,6 +366,35 @@ FirmwareUpdate_bootloaderLoop(const uint32_t timeout) {
     return retVal;
 }
 
+static bool
+FirmwareUpdate_sendStringWithCrc(uint8_t* string, size_t size) {
+
+    bool success = false;
+    bool null_terminator_exist = false;
+
+    for (size_t i = 0; i < size; ++i) {

```

Comment:
size_t is unsigned integer type. If we want to follow MISRA rules, we need to set size_t i = 0U